### PR TITLE
Refactor provision library.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: examples/ble_provision.toit [hunk 1]
```diff
@@ -71,21 +72,19 @@ USER-NAME ::= "wifiprov"
...
-  prov := provision.Provision.ble
-      service-name
-      security
+  prov := provision.Provision service-name --security-credentials=credentials
 
   prov.start
 
   note ::= """
-      Open the following URL in a browser:
+      For a QR code open the following URL in a browser:
```
Comment (kasperl) [inline examples/ble_provision.toit:87]:
```suggestion
      For a QR code, open the following URL in a browser:
```

File: src/src_/scan.toit [hunk 2]
```diff
@@ -1,51 +1,71 @@
...
+    ap-list/List := []
+
+  close -> none:
+    critical-do:
+      stop-scanning_
+      super
 
-  compare-ap-by-rssi a/wifi.AccessPoint b/wifi.AccessPoint -> int:
+  static compare-ap-by-rssi_ a/wifi.AccessPoint b/wifi.AccessPoint -> int:
     return -(a.rssi.compare-to b.rssi)
```
Comment (kasperl) [inline src/src_/scan.toit:40]: Isn't this a bit weird? Wouldn't it be more common to do `b.rssi.compare-to a.rssi`?

File: src/src_/security.toit [hunk 3]
```diff
@@ -1,32 +1,45 @@
...
 
-  handshake data/ByteArray -> ByteArray:
-    resp-msg := null
+interface Security:
+  encrypt data/ByteArray -> ByteArray
+  decrypt data/ByteArray -> ByteArray
+  handle-handshake-request data/ByteArray -> protobuf.Message
+  version -> int
 
+/** This security mode 0 doesn't encrypt/decrypt. */
```
Comment (kasperl) [inline src/src_/security.toit:31]: Is there any other security mode 0? Maybe just `/** Security mode 0 ...`.

File: src/src_/security.toit [hunk 4]
```diff
@@ -46,6 +57,7 @@ class Security0_ implements Security:
   version -> int:
     return SecSchemeVersion-SecScheme0
 
+/** This security mode 2 use SRP6a + AES-GCM. */
```
Comment (kasperl) [inline src/src_/security.toit:60]: Drop `This` unless there are other security 2 modes.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/src_/scan.toit (hunk 2 above)
Comment (floitsch) [reply src/src_/scan.toit:40]: I think I actually prefer this approach. Makes it clear that it is inverted.

File: src/src_/security.toit (hunk 3 above)
Comment (floitsch) [reply src/src_/security.toit]: done.

File: src/src_/security.toit (hunk 4 above)
Comment (floitsch) [reply src/src_/security.toit]: done.

**Pull request merged**